### PR TITLE
[newrelic-logging] Add support for Windows containers

### PR DESCRIPTION
**Pull request by jsubirat**
#### Is this a new chart
No.

#### What this PR does / why we need it:
The PR adds support to ship logs from Windows containers. This PR includes the changes contributed by @andrew-lozoya in this PR: https://github.com/newrelic/helm-charts/pull/170, but simplified so that we only need to install one Helm chart both for Windows and Linux.

#### Special notes for your reviewer:
- The main introduced change is the `daemonset-windows.yml` file. Some observations:
  - It creates **multiple** `DaemonSet`s, because we need a separate `DaemonSet` for each Windows version that we need to support. Right now, we only support Windows Server 2019 LTSC, therefore it will only create one `DaemonSet`.
  - It uses the same version of the `newrelic-fluent-bit-output` Docker image as the Linux Helm chart, but note that the image tag contains a suffix as the [Linux](https://hub.docker.com/layers/newrelic/newrelic-fluentbit-output/1.9.2/images/sha256-cf37c73e985875a65516317bb736737d71ca735602a19fdbdcfe68e9e8caddd5?context=explore) and [Windows](https://hub.docker.com/layers/newrelic/newrelic-fluentbit-output/1.9.2-windows-ltsc-2019/images/sha256-488c71004c6a5618c83e042bfac8214d48c4e0a830d55410712b43de3384e3b6?context=explore) images are different.
  - We ensure that the `DaemonSet` will only be deployed to nodes having the `kubernetes.io/os=windows` and `node.kubernetes.io/windows-build=SOME_BUILD` by using a `nodeSelector`. Note that we need to constraint this because **Windows containers can only be executed on hosts running the exact same Windows version and build number**. This limitation justifies many of the changes in this PR.
- We needed to un-split the `ConfigMap` into a single `fluent-bit.conf` file, because Fluent Bit running on a Windows container does not support the `@INCLUDE` directive. This way, the same `ConfigMap` can be reused across the two `DaemonSets` (Linux and Windows).
- As a result, we've had to add the `FB_DB` environment variable in order to provide different DB paths for Windows and Linux. Reminder: this is the DB where Fluent Bit will store up to which offset it has read from each log file. This file needs to be mounted _outside_ of any container (in the _host_ filesystem) in order to avoid losing this data across container redeployments.
- We now provide flags to enable/disable entirely the support for Linux and/or Windows. This is specially targeted for those users only having Linux machines that don't need to create the Windows `DaemonSet`.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-logging/templates/daemonset-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            - C:\fluent-bit\bin\out_newrelic.dll
+          volumeMounts:
+            - mountPath: C:\fluent-bit\etc
+              name: fluent-bit-config
+            - mountPath: C:\var\log
+              name: varlog
+              # We need to also mount this because the logs in C:\var\logs are actually symlinks to C:\ProgramData.
+              # So, in order to be able to read these logs, the reading process needs to also have access to C:\ProgramData.
+            - mountPath: C:\ProgramData
+              name: progdata
```
Comment (danybmx) [inline charts/newrelic-logging/templates/daemonset-windows.yaml:108]: 👍🏻 Nice finding!

**Review by danybmx — CHANGES_REQUESTED** [review]
Hey @jsubirat and @andrew-lozoya nice work! kudos!! 👏🏻 👏🏻 

I've just left a comment on values file for `enableWindows` that I think should be changed, if not, ping me and I'll approve this.

Thanks both for take care of windows containers!! 🪟🪟

File: charts/newrelic-logging/templates/configmap.yaml [hunk 2]
```diff
@@ -41,10 +35,9 @@ data:
     [FILTER]
         Name           kubernetes
         Match          kube.*
-        Kube_URL       https://kubernetes.default.svc:443
+        Kube_URL       https://kubernetes.default.svc.cluster.local:443
```
[the remark this answers is not in the document]
Comment (jsubirat) [reply charts/newrelic-logging/templates/configmap.yaml:38]: It took me a while to figure that out! Actually, I'm going to put this as a comment, thanks Dani!

File: charts/newrelic-logging/values.yaml [hunk 3]
```diff
@@ -42,6 +45,21 @@ image:
   # pullSecrets:
     # - name: regsecret
 
+# By default, the Linux DaemonSet will always be deployed, while the Windows DaemonSet(s) won't.
+enableLinux: true
+enableWindows: true
```
[the remark this answers is not in the document]
Comment (jsubirat) [reply charts/newrelic-logging/values.yaml:50]: Indeed, this is a leftover from my tests, I will be reverting this. Thanks!

File: charts/newrelic-logging/templates/daemonset-windows.yaml (hunk 1 above)
Comment (jsubirat) [reply charts/newrelic-logging/templates/daemonset-windows.yaml:108]: @andrew-lozoya 👏

**Review by danybmx — APPROVED** [review]
kudos!! 🏅🏅

File: charts/newrelic-logging/templates/daemonset-windows.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+{{ toYaml $.Values.image.pullSecrets | indent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ template "newrelic-logging.name" $ }}
+          image: "{{ $.Values.image.repository }}:{{ $.Values.image.tag | default $.Chart.AppVersion  }}-{{ .imageTagSuffix }}"
+          imagePullPolicy: "{{ $.Values.image.pullPolicy }}"
+          securityContext: {}
+          env:
+            - name: ENDPOINT
+              {{- if (include "newrelic.nrStaging" $) }}
```
Comment (tejunior) [inline charts/newrelic-logging/templates/daemonset-windows.yaml:53]: Do we need to expose staging? Maybe we can expose only US and EU

**Review by tejunior — APPROVED** [review]

File: charts/newrelic-logging/templates/daemonset-windows.yaml (hunk 4 above)
Comment (jsubirat) [reply charts/newrelic-logging/templates/daemonset-windows.yaml:53]: We are also doing this in the Linux daemonset, we can handle this, if needed, in a separate PR.

File: charts/newrelic-logging/templates/daemonset.yaml [hunk 5]
```diff
@@ -122,10 +125,16 @@ spec:
                     values:
                       - fargate
       {{- end }}
-      {{- if $.Values.nodeSelector }}
       nodeSelector:
+        {{- if $.Values.nodeSelector }}
```
Comment (paologallinaharbur) [inline charts/newrelic-logging/templates/daemonset.yaml:129]: I do not think that $ is needed here since we are not inside a loop

Comment (paologallinaharbur) [reply charts/newrelic-logging/templates/daemonset.yaml:129]: also in the if next to it is the same

Comment (jsubirat) [reply charts/newrelic-logging/templates/daemonset.yaml]: Indeed, fixed!

**Pull request merged**